### ISSUE TITLE
There should really be an indicator of progress of some kind. 

### ISSUE THREAD

**Issue by happysmash27** [ponchio/untrunc#197]
Untrunc finishes eventually, but often takes several hours to complete for each video (I ended up with a bunch of corrupted videos for some reason, ~6 hours each). Right now, I have about 5 untruncs which have been running for about 17 hours straight. When will they finish? I don't know, because there is no indicator of progress, not a file increasing a size, or logging which segment it is on, or anything.

**Comment by ponchio** [comment]
You are right, most probably though untrunc, failed and got stuck in a loop.
I am working on the logging (apt name, it seems) branch that has better
error reporting and some improved algorithms.
It's also way faster.

I think it's stable enough now and I should move that to the master. Would
you try?

In case it fails to recover your video, I would like to have a look at it
and possibly improve the code.

**Comment by mlindgren** [comment]
Same issue here. Untrunc has been running on a video of mine for two days straight. I'm not sure if it's stuck in a loop or just taking a long time - the video in question is 18gb and should be around 6 hours long, so I'm not sure how long that's expected to take.

I'd be happy to try the logging branch, but the build instructions for master don't work on the logging branch. Could you provide instructions for building logging?

**Comment by ponchio** [comment]
It's stuck in a loop, most probably, here are the modified instructions
(differences are clone, checkout logging and some more files to be
compiled).

 mkdir logging
logging/
wget https://libav.org/releases/libav-12.3.tar.xz
git clone https://github.com/ponchio/untrunc.git
cd untrunc/
git checkout logging
cd ..
tar -xJf libav-12.3.tar.xz -C untrunc/
cd untrunc/
cd libav-12.3/
./configure --disable-yasm
make
cd ..
g++ -o untrunc -I./libav-12.3 file.cpp main.cpp track.cpp atom.cpp mp4.cpp
log.cpp codec*.cpp -L./libav-12.3/libavformat -lavformat
-L./libav-12.3/libavcodec -lavcodec -L./libav-12.3/libavresample
-lavresample -L./libav-12.3/libavutil -lavutil -lpthread -lz -lbz2

in case untrunc fails again, you can send me the files (or even a sample),
and I will have a look at them.

**Comment by mlindgren** [comment]
Thanks for the quick response. I tried using @anthwlock's fork instead (https://github.com/anthwlock/untrunc) and it was able to recover the file successfully. It only took about 20-30 minutes, too. Maybe the changes from that fork can be merged in here?
